src/pages/previous.js

```diff
@@ -0,0 +1,152 @@
+import NavBar from "../components/navBar";
+import Chevron from "../assets/chevron_down.svg";
+import { GET } from "../utils/fetch";
+import { useEffect, useState } from "react";
+
+async function getQuestions() {
+  const weekly_questions = await GET("weekly_questions");
+  const questions = [];
+  for (const value in weekly_questions) {
+    questions.push({
+      id: weekly_questions[value].question_id,
+      body: weekly_questions[value].question.body,
+      time: new Date(weekly_questions[value].week),
+    });
+  }
+  return questions;
+}
+
+const options = {
+  month: "long",
+};
+
+function Previous() {
+  const [questions, setQuestions] = useState([]);
+  const [answers, setAnswers] = useState({});
+  const [answersVisibility, setAnswersVisibility] = useState({});
+
+  useEffect(() => {
+    getQuestions().then((response) => {
+      setQuestions(response);
+    });
+  }, []);
+
+  const viewAnswers = (questionId) => {
+    setAnswersVisibility((prevVisibility) => ({
+      ...prevVisibility,
+      [questionId]: !prevVisibility[questionId],
+    }));
+
+    getAll(questionId);
+  };
+
+  async function getAll(questionId) {
+    const answers = await GET("questions/" + questionId + "/answers");
+    await Promise.all(
+      answers.map((answer) => {
+        return new Promise((res) => {
+          GET("users/" + answer["user_id"].toString()).then((user) =>
+            res({ user, body: answer.body, time: new Date(answer.created_at) })
+          );
+        });
+      })
+    ).then((values) => {
+      const data = values.map((value) => {
+        return {
```

Comment:
Instead of having to do this, why not change the API so it returns the data in the format the FE is expecting?